main.java

```diff
@@ -0,0 +1,80 @@
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
+
+public class main {
+    // Swapping two elements function
+    public static void swap(String[] array, int c, int j) {
+        String temp = array[c];
+        array[c] = array[j];
+        array[j] = temp;
+    }
+
+    // Initiate last elment as pivot, insert the pivot elment in the array in a sorted manner
+    public static int partition(String[] arr, int left, int right) {
+        // Setting pivot
+        String pivot = arr[left];
```

Comment:
change the pivot to the rightmost element ya, the python version used the rightmost one)